src/main/java/com/fasterxml/jackson/core/json/JsonReadContext.java

```diff
@@ -61,6 +61,7 @@ public JsonReadContext(JsonReadContext parent, DupDetector dups, int type, int l
         _lineNr = lineNr;
         _columnNr = colNr;
         _index = -1;
+        _nestingDepth = parent == null ? 0 : parent._nestingDepth + 1;
```

Comment:
After merging, I may add an overload constructor to avoid null check, called from `createXxxContext()`, can mark this as deprecated.

Reply:
Done.